language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/ContractCompanion.java

```diff
@@ -44,7 +46,8 @@ protected ContractCompanion(
       String templateClassName,
       Identifier templateId,
       Function<String, Id> newContractId,
-      Function<DamlRecord, Data> fromValue) {
+      Function<DamlRecord, Data> fromValue,
+      List<ChoiceMetadata<Data, ?, ?>> choices) {
```

Comment:
Here's what I suggest we do with this thing:

1. add a `Marker` type parameter to `ContractTypeCompanion`. For `InterfaceCompanion` it should be `I`, for `ContractCompanion` it should be `Data`
2. Pass `choices` along to `super` (substituting `Marker` for `Data` in the type)
3. transform it to the instance variable `public final Map<String, ChoiceMetadata<Marker, ?, ?>> choices` (see `ContractDecoder` for an example that makes it safe and immutable)

That would be reasonable for this PR; for interfaces we can either include them in this PR or do a separate PR to supply them.